Sources/SpeziBluetooth/Bluetooth.swift

```diff
@@ -6,148 +6,336 @@
+    private let deviceConfigurations: Set<DiscoveryConfiguration>
+
+    private var logger: Logger {
+        Self.logger
+    }
+
+
+    /// Represents the current state of Bluetooth.
     public var state: BluetoothState {
         bluetoothManager.state
     }
-    
-    
-    /// Initializes the Bluetooth module with provided services.
-    ///
-    /// - Parameters:
-    ///   - services: List of Bluetooth services to manage.
-    public init(services: [BluetoothService]) {
-        bluetoothManager = BluetoothManager(services: services)
+
+    /// Whether or not we are currently scanning for nearby devices.
+    public var isScanning: Bool {
+        bluetoothManager.isScanning
     }
-    
-    /// Default initializer with no services specified.
-    public required convenience init() {
-        self.init(services: [])
+
+    @_documentation(visibility: internal)
+    public var hasConnectedDevices: Bool {
+        connectedDevicesModel.hasConnectedDevices
     }
-    
-    /// Sends a ByteBuffer to the connected Bluetooth device.
+
+
+    @MainActor private var nearbyDevices: [UUID: BluetoothDevice] = [:]
+
+    /// Stores the connected device instance for every configured ``BluetoothDevice`` type.
+    @Model @ObservationIgnored  private var connectedDevicesModel = ConnectedDevices()
+    /// Injects the ``BluetoothDevice`` instances from the `ConnectedDevices` model into the SwiftUI environment.
+    @Modifier @ObservationIgnored private var devicesInjector: ConnectedDevicesEnvironmentModifier
+
+
+    /// Configure the Bluetooth Module.
+    ///
+    /// Configures the Bluetooth Module with the provided set of ``DiscoveryConfiguration``s.
+    /// Below is a short code example on how you would discover a `ExampleDevice` by its advertised service id.
+    ///
+    /// ```swift
+    /// Bluetooth {
+    ///     Discover(ExampleDevice.self, by: .advertisedService("..."))
+    /// }
+    /// ```
     ///
     /// - Parameters:
-    ///   - byteBuffer: Data in ByteBuffer format to send.
-    ///   - service: UUID of the Bluetooth service.
-    ///   - characteristic: UUID of the Bluetooth characteristic.
-    public func write(_ byteBuffer: inout ByteBuffer, service: CBUUID, characteristic: CBUUID) async throws {
-        guard let data = byteBuffer.readData(length: byteBuffer.readableBytes) else {
+    ///   - minimumRSSI: The minimum rssi a nearby peripheral must have to be considered nearby.
+    ///   - advertisementStaleInterval: The time interval after which a peripheral advertisement is considered stale
+    ///     if we don't hear back from the device. Minimum is 1 second.
+    ///   - devices:
+    public init(
+        minimumRSSI: Int = BluetoothManager.Defaults.defaultMinimumRSSI,
+        advertisementStaleInterval: TimeInterval = BluetoothManager.Defaults.defaultStaleTimeout,
+        @DiscoveryConfigurationBuilder _ devices: () -> Set<DiscoveryConfiguration>
```

Comment:
I think it would be technically feasible with only a few changes to support something like that. It might need some more work if you want to do that within an ongoing scan but still possible. Additional state might need to be handled if you want to support removing certain discovery configurations again.

However, I would propose to not implement this (for now). Device implementations are static anyways (you will need to declare your Swift types). While this might not fully work when you use the auto connect feature, you can just choose to hide certain device types if you don't need them.

Short term, it might be the easiest to allow to specify a Device type for the `scanNearbyDevices` modifier (and methods). This covers most of the use cases and avoids all the unnecessary state handling on the `BluetoothManager` level.

Should I create an issue for that?